From 4dce3b50667221ef2af573a0244cdc6ad73ae53a Mon Sep 17 00:00:00 2001
From: Honggang Li <honli@redhat.com>
Date: Tue, 12 Feb 2019 13:35:40 +0800
Subject: [PATCH] fix a compilation issue
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

cc   -Wall -Werror  -fpic -fPIC -D_GNU_SOURCE -funwind-tables   -O3 -g3  -DPSM_USE_SYS_UUID  -DNVALGRIND -Wno-maybe-uninitialized -Wno-implicit-function-declaration -I. -I/home/honli/upstream/psm/include -I/home/honli/upstream/psm/mpspawn -I/home/honli/upstream/psm/include/linux-x86_64  -c psm_ep.c -o psm_ep.o
In function ‘psmi_parse_devices’,
    inlined from ‘__psm_ep_open’ at psm_ep.c:1003:16:
psm_ep.c:1352:5: error: ‘strncpy’ output truncated before terminating nul copying as many bytes from a string as its length [-Werror=stringop-truncation]
 1352 |     strncpy(e, devstring, len-1);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
psm_ep.c: In function ‘__psm_ep_open’:
psm_ep.c:1341:11: note: length computed here
 1341 |     len = strlen(devstring)+1;
      |           ^~~~~~~~~~~~~~~~~

Signed-off-by: Honggang Li <honli@redhat.com>
---
 psm_ep.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/psm_ep.c b/psm_ep.c
index 4114fc8..95a69bf 100644
--- a/psm_ep.c
+++ b/psm_ep.c
@@ -1349,7 +1349,7 @@ psmi_parse_devices(int devices[PTL_MAX_INIT], const char *devstring)
 
     b_new = (char *) devstr;
     e = b_new + len;
-    strncpy(e, devstring, len-1);
+    strncpy(e, devstring, len-2);
     e[len-1] = '\0';
     ee = e + len;
     i = 0;
-- 
2.20.1

